CDCgov/cdc-open-viz
# Refactor [DEV-9285] top suffix behavior and implementation adjustment

## [DEV-9285](https://websupport-cdc.msappproxy.net/browse/DEV-9285)
Two things happening here:

### 1. Refactor top-suffix-only logic so it overflows if a space is present in the suffix.
So:
"5lbs": pushes to the left
"5lbs per meter":  overflows
"5 units": overflows

### 2. Remove getTextWidth from context and import instead

## Testing Steps

In the editor:
1. Create a Bar or Line chart
2. Open the "Left Value Axis" panel
3. Select the checkbox labeled "Only Show Prefix/Suffix" 
4. Add a Suffix without a space (i.e. something like 'lbs')
5. Observe that the suffix pushes the tick labels to the left, to the left
6. Add a suffix with a space (i.e. something like " units" or "lbs of something")
7. Observe that the suffix now overflows to the right instead

## Self Review

- I have added testing steps for reviewers
- I have commented my code, particularly in hard-to-understand areas
- My changes generate no new warnings
- New and existing unit tests are passing

## Screenshots (if applicable)

<img width="634" alt="Screenshot 2024-10-21 at 12 19 07 PM" src="https://github.com/user-attachments/assets/670f91d4-f073-4c82-ab9f-d367a4fa4313">
<img width="656" alt="Screenshot 2024-10-21 at 12 19 01 PM" src="https://github.com/user-attachments/assets/6490bc89-019d-47ba-8619-e0655ff7c8cb">
